fggs/factors.py

```diff
@@ -55,8 +55,10 @@ def __init__(self, doms, weights):
           - or a torch.Tensor,
           - or a fggs.indices.PatternedTensor"""
         
-        if not all(isinstance(d, domains.FiniteDomain) for d in doms):
-            raise TypeError('FiniteFactor can only be applied to FiniteDomains')
+        if not all(isinstance(d, domains.FiniteDomain) or
+                   isinstance(d, domains.CompactDomain)
```

Comment:
isinstance(d, (domains.FiniteDomain, domains.CompactDomain))

(or make both of them inherit from the same parent class)

Reply:
> isinstance(d, (domains.FiniteDomain, domains.CompactDomain))
> 
> (or make both of them inherit from the same parent class)

I turned `size` into an abstract method in `Domain`.